File: Algorithms/Easy/1.Two-Sum/TwoSum.py
# Time complexity O(n)
# Space complexity O(n)
class Solution(object):
    def twoSum(self, nums, target):
        subtr = {}

        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in subtr:
                return [i, subtr[diff]]
            
            subtr[nums[i]] = i

File: Algorithms/Easy/1.Two-Sum/TwoSum.go
package easy

// Time complexity O(n)
// Space complexity O(n)
func twoSum(nums []int, target int) []int {
	hashMap := make(map[int]int, len(nums))

	for i := 0; i < len(nums); i++ {
		diff := target - nums[i]
		if _, ok := hashMap[diff]; ok {
			return []int{i, hashMap[diff]}
		}

		hashMap[nums[i]] = i
	}

	return nil
}
